lib/spack/spack/detection/path.py

```diff
@@ -252,13 +247,18 @@ def find(
         patterns = self.search_patterns(pkg=pkg_cls)
```

Comment:
Can you update the docstring of this method to reflect the new behavior?

(Originally, if `initial_guess = []`, then this would still add default paths via `path_hints`, now it doesn't)

Reply:
Addressed and pushed.